src/lib.rs

```diff
@@ -88,27 +89,35 @@ fn json_set(ctx: &Context, args: Vec<String>) -> RedisResult {
     let path = backwards_compat_path(args.next_string()?);
     let value = args.next_string()?;
 
-    let set_option = args
-        .next()
-        .map(|op| match op.to_uppercase().as_str() {
-            "NX" => Ok(SetOptions::NotExists),
-            "XX" => Ok(SetOptions::AlreadyExists),
-            _ => Err(RedisError::Str("ERR syntax error")),
-        })
-        .transpose()?;
+    let mut format = Format::JSON;
```

Comment:
The idiomatic Rust style would avoid using mutable variables.
I previously refactored the original code to get rid of the mutables and use a single expression; here we're back with mutables again.

Reply:
that was right when we only parsed SetOptions but now we have another flag here

Reply:
Sure, but that doesn’t change the principle. In any case, it’s not worth arguing about.

Reply:
No arguing, if you have a cleaner way please suggest

Reply:
👍 I'm OK with merging this now and not spending more time on it; maybe as part of a future code cleanup.